cmd/juju/resource/charmresources.go

```diff
@@ -0,0 +1,196 @@
+	if resourceLister == nil {
+		resourceLister = b
+	}
+	b.resourceLister = resourceLister
+}
+
+func (c *baseCharmResourcesCommand) baseInfo() *cmd.Info {
+	return &cmd.Info{
+		Args:    "<charm>",
+		Purpose: "Display the resources for a charm in the charm store.",
+		Doc:     charmResourcesDoc,
+	}
+}
+
+func (c *baseCharmResourcesCommand) setBaseFlags(f *gnuflag.FlagSet) {
+	c.ModelCommandBase.SetFlags(f)
+	defaultFormat := "tabular"
+	c.out.AddFlags(f, defaultFormat, map[string]cmd.Formatter{
+		"tabular": FormatCharmTabular,
+		"yaml":    cmd.FormatYaml,
+		"json":    cmd.FormatJson,
+	})
+	f.StringVar(&c.channel, "channel", "stable", "the charmstore channel of the charm")
+}
+
+func (c *baseCharmResourcesCommand) baseInit(args []string) error {
+	if len(args) == 0 {
+		return errors.New("missing charm")
+	}
+	c.charm = args[0]
+
+	if err := cmd.CheckEmpty(args[1:]); err != nil {
+		return errors.Trace(err)
+	}
+	return nil
+}
+
+func (c *baseCharmResourcesCommand) baseRun(ctx *cmd.Context) error {
+	// TODO(ericsnow) Adjust this to the charm store.
+
+	charmURLs, err := resolveCharms([]string{c.charm})
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	charms := make([]charmstore.CharmID, len(charmURLs))
+	for i, id := range charmURLs {
+		charms[i] = charmstore.CharmID{URL: id, Channel: csparams.Channel(c.channel)}
+	}
+
+	resources, err := c.resourceLister.ListResources(charms)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if len(resources) != 1 {
+		return errors.New("got bad data from charm store")
+	}
+	res := resources[0]
+
+	if len(res) == 0 && c.out.Name() == "tabular" {
+		ctx.Infof("No resources to display.")
+		return nil
+	}
+
+	// Note that we do not worry about c.CompatVersion
+	// for show-charm-resources...
+	formatter := newCharmResourcesFormatter(resources[0])
+	formatted := formatter.format()
+	return c.out.Write(ctx, formatted)
+}
+
+var charmResourcesDoc = `
+This command will report the resources for a charm in the charm store.
+
+<charm> can be a charm URL, or an unambiguously condensed form of it,
+just like the deploy command. So the following forms will be accepted:
+
+For cs:trusty/mysql
+  mysql
+  trusty/mysql
+
+For cs:~user/trusty/mysql
+  cs:~user/mysql
+
+Where the series is not supplied, the series from your local host is used.
+Thus the above examples imply that the local series is trusty.
+`
+
+// ListCharmResources implements CharmResourceLister by getting the charmstore client
+// from the command's ModelCommandBase.
+func (c *baseCharmResourcesCommand) ListResources(ids []charmstore.CharmID) ([][]charmresource.Resource, error) {
+	bakeryClient, err := c.BakeryClient()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	client, err := charmstore.NewCustomClient(bakeryClient, nil)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	return client.ListResources(ids)
+}
+
+func resolveCharms(charms []string) ([]*charm.URL, error) {
```

Comment:
Not your doing I know but this is kinda silly given that there can only be exactly one charm. This could easily go and `resolveCharm` could be called directly instead. This would simplify `baseRun` as well.